functions/localhost_or_hosts_with_pe_profile.pp

```diff
@@ -12,21 +12,32 @@
 function puppet_metrics_dashboard::localhost_or_hosts_with_pe_profile(
   String $profile,
 ) >> Array {
-  if $settings::storeconfigs {
-    $_profile = capitalize($profile)
-    $hosts = puppetdb_query("resources[certname] {
-      type = 'Class' and
-      title = 'Puppet_enterprise::Profile::${_profile}' and
-      nodes { deactivated is null and expired is null }
-    }").map |$nodes| { $nodes['certname'] }
+  # PE 2019.5 or newer with Puppet Server 6.9.2 or newer, or
+  # PE 2018.1.13 or newer PE 2018 with Puppet Server 5.3.12 or newer
+  # As per: https://puppet.com/docs/pe/latest/component_versions_in_recent_pe_releases.html
+  $puppetdb_no_remote_metrics = ((versioncmp($facts['puppetversion'], '6.14.0') >= 0) or ((versioncmp($facts['puppetversion'], '5.5.19') >= 0) and (versioncmp($facts['puppetversion'], '6.0.0') < 1)))
+  if ($profile == 'puppetdb') and $puppetdb_no_remote_metrics {
+    if $facts['puppet_server'] == $trusted['certname'] {
+      $hosts = ['localhost']
+    } else {
+      $hosts = []
```

Comment:
Thank you for this, I know its an "unnecessary" assignment for a return value, but it makes it easier to read!